### PR TITLE
Vault test refactor

### PR DESCRIPTION
**Pull request by dtribble**
refs: #4568 

## Description

Refactor test support into shared module
* add test/support.js for test helpers
* consolidate `makeBundle` and `setupZoeForTest`
* make `setupZoeForTest` sync and return promises
* add bootstrap `space` setup helper

Refactor vault tests:
* add `test.before` for shared setup
* use `t.context` pervasively for shared and default config
* change methods to override `t.context` before invoking `setupServices`
* use `econ-behaviors` directly
* move `addVaultType` into shared setup
* eliminate governance terms

### Security Considerations

N/A

### Documentation Considerations

N/A

### Testing Considerations

This simplifies setup for RUN tests, and dramatically reduces duplicate code and setup in vault tests especially. It also reduces some redundant work in tests like creation of a new Zoe for each independent test.  Additional shared overhead is now possible via `test.before`

### CODE REVIEW

File: packages/run-protocol/src/makeTracer.js [hunk 1]
```diff
@@ -17,8 +17,12 @@ const makeTracer = (name, enable = true) => {
       return infoTick;
     }
     default: {
-      const debugTick = (...args) => {
-        console.log(key, (debugCount += 1), ...args);
+      const debugTick = (optLog, ...args) => {
+        if (optLog.log) {
```
Comment (turadg) [inline packages/run-protocol/src/makeTracer.js:21]: I hope we find time after MN-1 to rationalize logging. I don't think we have a ticket yet for that per se. https://github.com/Agoric/agoric-sdk/issues/1318 is to do research.

File: packages/run-protocol/src/vaultFactory/vaultManager.js [hunk 2]
```diff
@@ -157,6 +157,7 @@ export const makeVaultManager = (
         liquidationInProgress = false;
         // XXX should notify interested parties
         console.error('liquidateAndRemove failed with', e);
+        throw e;
```
Comment (turadg) [inline packages/run-protocol/src/vaultFactory/vaultManager.js]: non-blocking: this is out of scope.

File: packages/run-protocol/src/vaultFactory/liquidation.js [hunk 3]
```diff
@@ -68,13 +68,14 @@ const liquidate = async (
     collateralBrand,
   );
 
-  // XXX problems with upgrade
```
Comment (turadg) [inline packages/run-protocol/src/vaultFactory/liquidation.js]: gratuitous removal. this is still true.

**Review by turadg — APPROVED** [review]
The PR title omits that `src` is being refactored but the commits look good. I assume the PR will be merged with out squashing.

**Review by Chris-Hibbert — APPROVED** [review]

**Comment by dtribble** [comment]
@Chris-Hibbert I had included the new `manualPriceAuthority` and test changes with some cleanups here. Rather than separate them out, since they are really part of this overall set of changes, can you review the last commit and provide additional approval (if appropriate)?

File: packages/run-protocol/test/vaultFactory/test-vaultFactory.js [hunk 4]
```diff
@@ -535,11 +545,17 @@ test('price drop', async t => {
     AmountMath.make(aethBrand, 400n),
     'vault holds 11 Collateral',
   );
-  await manualTimer.tick();
+  trace(t, 'pa2', priceAuthority);
+
+  priceAuthority.setPrice(makeRatio(677n, runBrand, 900n, aethBrand));
+  trace(t, 'price dropped a little');
+  // await manualTimer.tick();
```
Comment (Chris-Hibbert) [inline packages/run-protocol/test/vaultFactory/test-vaultFactory.js:552]: any reason to leave these `tick()`s here long term?

**Review by Chris-Hibbert — APPROVED** [review]
These are improvements.

File: packages/run-protocol/test/vaultFactory/test-vaultFactory.js (hunk 4 above)
Comment (dtribble) [reply packages/run-protocol/test/vaultFactory/test-vaultFactory.js:552]: nope. I'll clean that up in a future PR.
